src/devices/Lcm1602a1/HD44780.cs

```diff
@@ -0,0 +1,559 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Gpio.Drivers;
+using System.Diagnostics;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Iot.Device.CharacterLCD
+{
+    /// <summary>
+    /// Supports LCD character displays compatible with the HD44780 LCD controller/driver.
+    /// Also supports serial interface adapters such as the MCP23008.
+    /// </summary>
+    /// <remarks>
+    /// The Hitatchi HD44780 was released in 1987 and set the standard for LCD controllers. Hitatchi does not make this chipset anymore, but
+    /// most character LCD drivers are intended to be fully compatible with this chipset. Some examples: Sunplus SPLC780D, Sitronix ST7066U,
+    /// Samsung KS0066U (and many more).
+    /// 
+    /// Some compatible chips extend the HD44780 with addtional pins and features. They are still fully compatible. The ST7036 is one example.
+    /// 
+    /// This implementation was drawn from numerous datasheets and libraries such as Adafruit_Python_CharLCD.
+    /// </remarks>
+    public class HD44780 : IDisposable
```

Comment:
Given that this is the main controller now, I would just go ahead and remove the Lcm1602a1 file from the PR.

Reply:
Whoops. I'm about to rename shortly as well.

Reply:
also was this supposed to be abstract? (given that you said you would provide LCM specific classes) or is the idea that this would be sort of a generic class, and then there would be device-specific sub classes?

Reply:
No, didn't want to force people to derive when it should (in theory) just work with any LCD layout.